systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java

```diff
@@ -99,106 +101,110 @@ public LogCollector(String testSuiteName, String testCaseName, KubeClient kubeCl
      * Core method which collects all logs from events, configs-maps, pods, deployment, statefulset, replicaset...in case test fails.
      */
     public synchronized void collect() {
-        final Set<String> namespaces = KubeClusterResource.getInstance().getMapWithSuiteNamespaces().get(this.testSuiteName);
+        Set<String> namespaces = KubeClusterResource.getMapWithSuiteNamespaces().get(this.collectorElement);
+        AtomicReference<CollectorElement> collectorElement = new AtomicReference<>();
+        collectorElement.set(CollectorElement.emptyElement());
+
+        if (namespaces == null) {
+            collectorElement.set(CollectorElement.createCollectorElement(this.collectorElement.getTestClassName()));
+            namespaces = KubeClusterResource.getMapWithSuiteNamespaces().get(collectorElement.get());
+        }
 
         if (namespaces != null) {
+            namespaces.add(clusterOperatorNamespace);
             // collect logs for all namespace related to test suite
-            namespaces.stream().forEach(namespace -> {
-                namespaceFile = new File(testCase + "/" + namespace);
+            namespaces.forEach(namespace -> {
+                if (collectorElement.get().isEmpty()) {
+                    namespaceFile = new File(this.testCase + "/" + namespace);
+                } else {
+                    namespaceFile = new File(this.testSuite +  "/" + namespace);
+                }
```

Comment:
Would it make sense to use ternary operator here?

Reply:
This is more readable.

Reply:
Yes, make sense.